Osiris/Hacks/InventoryChanger.cpp

```diff
@@ -0,0 +1,1728 @@
+#include "../SDK/FrameStage.h"
+#include "../SDK/GameEvent.h"
+#include "../SDK/GlobalVars.h"
+#include "../SDK/ItemSchema.h"
+#include "../SDK/Localize.h"
+#include "../SDK/LocalPlayer.h"
+#include "../SDK/ModelInfo.h"
+#include "../SDK/Panorama.h"
+#include "../SDK/PlayerResource.h"
+#include "../SDK/Platform.h"
+#include "../SDK/WeaponId.h"
+
+#include "../Helpers.h"
+
+constexpr auto isKnife(WeaponId id) noexcept
+{
+    return (id >= WeaponId::Bayonet && id <= WeaponId::SkeletonKnife) || id == WeaponId::KnifeT || id == WeaponId::Knife;
+}
+
+namespace StaticData
+{
+    enum class Type {
+        Sticker,
+        Glove,
+        Skin,
+        Music,
+        Collectible,
+        NameTag
+    };
+
+    struct GameItem {
+        GameItem(Type type, int rarity, std::size_t dataIndex, std::wstring&& name, std::string&& iconPath) noexcept;
+
+        bool isSticker() const noexcept { return type == Type::Sticker; }
+        bool isSkin() const noexcept { return type == Type::Skin; }
+        bool isGlove() const noexcept { return type == Type::Glove; }
+        bool isMusic() const noexcept { return type == Type::Music; }
+        bool isCollectible() const noexcept { return type == Type::Collectible; }
+        bool isNameTag() const noexcept { return type == Type::NameTag; }
+
+        Type type;
+        int rarity;
+        std::size_t dataIndex;
+
+        std::string name;
+        std::wstring nameUpperCase;
+        std::string iconPath;
+    };
+
+    struct Sticker {
+        Sticker(int stickerID) : stickerID{ stickerID } {}
+
+        int stickerID;
+    };
+
+    struct Glove {
+        Glove(int paintKit, WeaponId weaponId) : paintKit{ paintKit }, weaponId{ weaponId } {}
+
+        int paintKit;
+        WeaponId weaponId;
+    };
+
+    struct Skin {
+        Skin(int paintKit, WeaponId weaponId) : paintKit{ paintKit }, weaponId{ weaponId } {}
+
+        int paintKit;
+        WeaponId weaponId;
+    };
+
+    struct Music {
+        Music(int musicID) : musicID{ musicID } {}
+
+        int musicID;
+    };
+
+    struct Collectible {
+        Collectible(WeaponId weaponId, bool isOriginal) : weaponId{ weaponId }, isOriginal{ isOriginal } {}
+
+        WeaponId weaponId;
+        bool isOriginal;
+    };
+
+    static std::vector<GameItem> gameItems;
+    static std::vector<Sticker> stickers;
+    static std::vector<Glove> gloves;
+    static std::vector<Skin> skins;
+    static std::vector<Music> music;
+    static std::vector<Collectible> collectibles;
+}
+
+struct StickerConfig {
+    int stickerID = 0;
+    float wear = 0.0f;
+};
+
+struct DynamicSkinData {
+    float wear = 0.0f;
+    int seed = 1;
+    std::array<StickerConfig, 5> stickers;
+    std::string nameTag;
+};
+
+struct DynamicGloveData {
+    float wear = 0.0f;
+    int seed = 1;
+};
+
+static std::vector<DynamicSkinData> dynamicSkinData;
+static std::vector<DynamicGloveData> dynamicGloveData;
+
+constexpr auto BASE_ITEMID = 1152921504606746975;
```

Comment:
like i understand all items you will add will be on top with this method, even if u receive other item from game. there's other way to do it

Reply:
This is method guarantees that there will be no collision between real and fake item IDs.